lib/cc/cli/analyze.rb

```diff
@@ -70,14 +70,19 @@ def exclude_paths
 
       def engines
         @engines ||= config.engine_names.map do |engine_name|
-          Engine.new(
-            engine_name,
-            registry_entry(engine_name),
-            path,
-            engine_config(engine_name),
-            SecureRandom.uuid
-          )
-        end
+          entry = registry_entry(engine_name)
+          if entry.nil?
+            warn("unknown engine name: #{engine_name}")
```

Comment:
We have a [fatal method](https://github.com/codeclimate/codeclimate/blob/f3451ba424525ad60ef3508f81fd2e7e866bb751/lib/cc/cli/command.rb#L35-L38) that I think would be a better fit here